pkg/kubelet/allocation/allocation_manager_test.go

```diff
@@ -1020,6 +1020,271 @@ func TestHandlePodResourcesResizeWithSwap(t *testing.T) {
+			ipprFeatureGate:                true,
+			initialAllocatedResourcesState: map[types.UID]v1.ResourceList{pod1UID: cpu1Mem1G},
+			currentActivePods:              []*v1.Pod{pod1Small},
+			podToAdd:                       pod2Large,
+			admitFunc: func(attrs *lifecycle.PodAdmitAttributes) lifecycle.PodAdmitResult {
+				cpuRequest := attrs.Pod.Spec.Containers[0].Resources.Requests.Cpu().Value()
+				if cpuRequest > 1 {
+					return lifecycle.PodAdmitResult{
+						Admit:   false,
+						Reason:  "OutOfcpu",
+						Message: fmt.Sprintf("not enough CPUs available for pod %s/%s, requested: %d, available:1", attrs.Pod.Namespace, attrs.Pod.Name, cpuRequest),
+					}
+				}
+				return lifecycle.PodAdmitResult{Admit: true}
+			},
+			expectAdmit:             false,
+			admissionFailureReason:  "OutOfcpu",
+			admissionFailureMessage: "not enough CPUs available for pod ns2/pod2, requested: 2, available:1",
+			// allocated resources not modified
+			expectedAllocatedResourcesState: map[types.UID]v1.ResourceList{pod1UID: cpu1Mem1G},
+		},
+		{
+			name:                           "IPPR Disabled - New pod not admitted due to insufficient resources",
+			ipprFeatureGate:                false,
+			initialAllocatedResourcesState: map[types.UID]v1.ResourceList{pod1UID: cpu1Mem1G},
+			currentActivePods:              []*v1.Pod{pod1Small},
+			podToAdd:                       pod2Large,
+			admitFunc: func(attrs *lifecycle.PodAdmitAttributes) lifecycle.PodAdmitResult {
+				cpuRequest := attrs.Pod.Spec.Containers[0].Resources.Requests.Cpu().Value()
+				if cpuRequest > 1 {
+					return lifecycle.PodAdmitResult{
+						Admit:   false,
+						Reason:  "OutOfcpu",
+						Message: fmt.Sprintf("not enough CPUs available for pod %s/%s, requested: %d, available:1", attrs.Pod.Namespace, attrs.Pod.Name, cpuRequest),
+					}
+				}
+				return lifecycle.PodAdmitResult{Admit: true}
+			},
+			expectAdmit:             false,
+			admissionFailureReason:  "OutOfcpu",
+			admissionFailureMessage: "not enough CPUs available for pod ns2/pod2, requested: 2, available:1",
+			// allocated resources not modified
+			expectedAllocatedResourcesState: map[types.UID]v1.ResourceList{pod1UID: cpu1Mem1G},
+		},
+		{
+			name:                            "IPPR Enabled - pod resize request feasible. Request same as existing allocation",
+			ipprFeatureGate:                 true,
+			initialAllocatedResourcesState:  map[types.UID]v1.ResourceList{pod1UID: cpu1Mem1G, pod2UID: cpu1Mem1G},
+			currentActivePods:               []*v1.Pod{pod1Small},
+			podToAdd:                        pod1Small,
+			admitFunc:                       nil,
+			expectAdmit:                     true,
+			expectedAllocatedResourcesState: map[types.UID]v1.ResourceList{pod1UID: cpu1Mem1G, pod2UID: cpu1Mem1G},
+		},
+		{
+			name:                           "IPPR Enabled - pod resize request feasible. Request different from current allocation. Admitted and allocated resources updated",
```

Comment:
I wouldn't describe this test as "request different from current allocation" - more like "current allocation not found for added pod", does that sound right? Suggest to update the wording accordingly.

I think what I think of when I read "request different from current allocation" is closer to the kubelet restart scenario and more useful to have:
- `initialAllocatedResourcesState` should have both `pod1` and `pod2` set to `cpu1Mem1G`
- `podToAdd` should be either `pod1Large` or `pod2Large`, so that the resource requests are different from the allocated resources
- after the call to `AddPod`, the allocation should remain unchanged, as the pod should be admitted based on the initialAllocatedResources

Could you add this case as well?